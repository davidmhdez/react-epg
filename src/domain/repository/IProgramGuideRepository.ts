import ProgramGuide from "../entity/ProgramGuide.ts";

export default interface IProgramGuideRepository {
    getProgramGuide(fromDate: Date, toDate: Date): Promise<ProgramGuide>
}