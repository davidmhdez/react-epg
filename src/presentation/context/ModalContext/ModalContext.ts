import {createContext} from "react";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type ModalContextValues = {
    isOpen: boolean
    openModal: () => void
    closeModal: () => void
    selectedChannelEvent: ChannelEvent | null
    selectChannelEvent: (channelEvent: ChannelEvent) => void
}

export const initialModalContextValues = {
    isOpen: false,
    selectedChannelEvent: null,
    closeModal: () => {},
    openModal: () => {},
    selectChannelEvent: () => {},
}

const ModalContext = createContext<ModalContextValues>(initialModalContextValues)

export default ModalContext