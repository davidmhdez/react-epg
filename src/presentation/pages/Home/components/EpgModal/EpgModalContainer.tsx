import EpgModal from "./EpgModal.tsx";
import container from "../../../../../inversify.config.ts";
import TYPES from "../../../../../domain/types/types.ts";
import GetProgramGuideUseCase from "../../../../../domain/interactor/ProgramGuide/GetProgramGuideUseCase.ts";
import {useQuery} from "react-query";

const EpgModalContainer = () => {
    const getProgramGuideUseCase = container.get<GetProgramGuideUseCase>(TYPES.GetProgramGuideUseCase);
    const getProgramGuide = () => getProgramGuideUseCase.getFullProgramGuide();

    const { data, isLoading } = useQuery('programGuide', getProgramGuide, {
        refetchOnWindowFocus: false,
        staleTime: 60000
    });

    return (
        <EpgModal
            isLoading={isLoading}
            programGuide={data ?? null}
        />
    );
};

export default EpgModalContainer;
