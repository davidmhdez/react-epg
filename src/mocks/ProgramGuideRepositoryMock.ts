import IProgramGuideRepository from "../domain/repository/IProgramGuideRepository.ts";
import ProgramGuide from "../domain/entity/ProgramGuide.ts";
import {injectable} from "inversify";
import "reflect-metadata"

@injectable()
export default class ProgramGuideRepositoryMock implements IProgramGuideRepository{
    getProgramGuide(fromDate: Date, toDate: Date): Promise<ProgramGuide> {
        return new Promise((resolve) => {
            resolve({
                from: fromDate,
                end: toDate,
                channels: [
                    {
                        id: "1",
                        name: "Test channel",
                        number: "3",
                        image: "",
                        events: [
                            {
                                id: "11",
                                name: "Test event",
                                duration: "01:30:00",
                                dateBegin: fromDate,
                                description: "Test description",
                                durationInMinutes: 90,
                                dateEnd: new Date(fromDate.getTime() + (90 * 60 * 1000))
                            }
                        ]
                    }
                ]
            })
        })
    }
}