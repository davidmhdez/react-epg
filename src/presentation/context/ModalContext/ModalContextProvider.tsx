import React, {ReactNode, useState} from "react";
import ModalContext from "./ModalContext.ts";
import {ChannelEvent} from "../../../domain/entity/Channel.ts";

type Props = {
    children: ReactNode
}

const ModalContextProvider: React.FC<Props> = ({children}) => {
    const [isOpen, setIsOpen] = useState(false);
    const [selectedChannelEvent, setSelectedChannelEvent] = useState<ChannelEvent | null>(null)
    const openModal = () => {
        setIsOpen(true);
        setSelectedChannelEvent(null);
    }
    const closeModal = () => {
        setIsOpen(false);
        setSelectedChannelEvent(null);
    }

    const selectChannelEvent = (channelEvent: ChannelEvent) => setSelectedChannelEvent(channelEvent);

    return (
        <ModalContext.Provider value={{
            isOpen,
            selectedChannelEvent,
            selectChannelEvent,
            openModal,
            closeModal
        }}>
            {children}
        </ModalContext.Provider>
    );
};

export default ModalContextProvider;
