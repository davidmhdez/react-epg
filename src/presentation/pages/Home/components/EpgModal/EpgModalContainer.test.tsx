import {render, waitFor, screen} from "@testing-library/react";
import ModalContext, {initialModalContextValues} from "../../../../context/ModalContext/ModalContext.ts";
import EpgModalContainer from "./EpgModalContainer.tsx";
import container from "../../../../../../inversify.config.ts";
import IProgramGuideRepository from "../../../../../domain/repository/IProgramGuideRepository.ts";
import TYPES from "../../../../../domain/types/types.ts";
import ProgramGuideRepositoryMock from "../../../../../mocks/ProgramGuideRepositoryMock.ts";
import {QueryClient, QueryClientProvider} from "react-query";
import "@testing-library/jest-dom"

describe('EpgModalContainer', () => {

    beforeEach(() => {
        container.rebind<IProgramGuideRepository>(TYPES.ProgramGuideRepository).to(ProgramGuideRepositoryMock);
    });

    test('should match with snapshot', async () => {
        const queryClient = new QueryClient();
        render(
            <QueryClientProvider client={queryClient}>
                <ModalContext.Provider value={initialModalContextValues}>
                    <EpgModalContainer/>
                </ModalContext.Provider>
            </QueryClientProvider>
        )

        await waitFor(()=>{
            expect(screen.getByTestId('epgTable')).toBeInTheDocument()
        })
    })
})