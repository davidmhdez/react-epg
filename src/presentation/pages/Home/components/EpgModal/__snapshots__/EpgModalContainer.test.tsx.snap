// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpgModalContainer should match with snapshot 1`] = `
<div>
  <div
    class="sc-aYaIB fEhQo"
  >
    <button
      class="sc-gEvDqW crccQQ"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          height="24"
          id="mask0_4409_5890"
          maskUnits="userSpaceOnUse"
          width="24"
          x="0"
          y="0"
        >
          <rect
            fill="#D9D9D9"
            height="24"
            width="24"
          />
        </mask>
        <g
          mask="url(#mask0_4409_5890)"
        >
          <path
            d="M6.40008 18.3077L5.69238 17.6L11.2924 12L5.69238 6.4L6.40008 5.6923L12.0001 11.2923L17.6001 5.6923L18.3078 6.4L12.7078 12L18.3078 17.6L17.6001 18.3077L12.0001 12.7077L6.40008 18.3077Z"
            fill="#0F265C"
          />
        </g>
      </svg>
    </button>
    <div
      class="sc-dAbbbq kLeOAP"
    />
    <div
      class="sc-eqUzNf bMwGHl"
      data-testid="epgModalSkeleton"
    >
      <div
        class="sc-fqkwJk cLpyiB"
      >
        <div
          class="sc-dcJtft cuVfsq"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton dayText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton hourText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton hourText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton hourText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton hourText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton hourText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton hourText"
              style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <div
          class="sc-iGgVNO iqwQsm"
        >
          <div
            class="sc-gsFSjX itFdMl"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton channelNumber"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton channelImage"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU fJRDwI"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU bydbjL"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU kHFOqR"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU TGBxm"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU haLdaA"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </div>
        <div
          class="sc-iGgVNO iqwQsm"
        >
          <div
            class="sc-gsFSjX itFdMl"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton channelNumber"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton channelImage"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU bydbjL"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU itGMYF"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU gSikKE"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU bydbjL"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU dTNDwL"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </div>
        <div
          class="sc-iGgVNO iqwQsm"
        >
          <div
            class="sc-gsFSjX itFdMl"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton channelNumber"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton channelImage"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU bVzwYM"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU dvSCew"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU fmxNre"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU hppeQH"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <div
            class="sc-kAycRU bXHmnt"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programTitle"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton programDuration"
                style="--base-color: #6C6C6CAD; --highlight-color: #7E7E7EAD;"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
